File: CS/DocumentExportWithTokenBasedAuthentication/Services/ReportResolver.cs
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.Web.WebDocumentViewer;
using T680906.Reports;

namespace T680906.Services
{
    public class ReportResolver : IWebDocumentViewerReportResolver
    {
        public XtraReport Resolve(string reportEntry)
        {
            if(reportEntry == "testReport") {
                return new XtraReport1();
            }
            return null;
        }
    }
}

File: VB/DocumentExportWithTokenBasedAuthentication/Services/ReportResolver.vb
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.Web.WebDocumentViewer
Imports T680906.Reports

Namespace T680906.Services
	Public Class ReportResolver
		Implements IWebDocumentViewerReportResolver

		Public Function Resolve(ByVal reportEntry As String) As XtraReport Implements IWebDocumentViewerReportResolver.Resolve
			If reportEntry = "testReport" Then
				Return New XtraReport1()
			End If
			Return Nothing
		End Function
	End Class
End Namespace